Move Jetpack Backup landing page link from selector to formatted header

delawski opened the pull request:
As part of the work related to p1HpG7-87V-p2, this PR moves the link to Jetpack Backup landing page from `ProductSelector` to `FormattedHeader`.

**Before**
<img width="557" alt="Screenshot 2019-12-10 at 13 02 41" src="https://user-images.githubusercontent.com/478735/70527959-5f170f80-1b4d-11ea-817d-ec434ef3c360.png">

**After**
<img width="563" alt="Screenshot 2019-12-10 at 13 02 08" src="https://user-images.githubusercontent.com/478735/70527918-4b6ba900-1b4d-11ea-89ec-cfe3ddea2991.png">

#### Changes proposed in this Pull Request

* Move Jetpack Backup landing page link from selector to formatted header

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to http://calypso.localhost:3000/plans/ and select a Jetpack site having no Jetpack Backup product
* Confirm that the link to the landing page is now located in the header and not inside the products selector.
* Purchase one of the Backup products and confirm that there's no link displayed on 
http://calypso.localhost:3000/plans/ for the Jetpack site.
* Remove the previously purchased Backup product and purchase one of the paid plans. Confirm that there's still no link displayed on http://calypso.localhost:3000/plans/ for the Jetpack site.

Fixes n/a

a8ci18n commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -434,6 +439,41 @@ export class PlansFeaturesMain extends Component {
 
+	getProductsSelectorSubHeader() {
+		const { purchases, siteSlug, translate } = this.props;
+		const baseCopy = translate( "Just looking for a backups? We've got you covered." );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Just looking for backups? We’ve got you covered.' )` *ES Score: 6*

delawski commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -434,6 +439,41 @@ export class PlansFeaturesMain extends Component {
+		return (
+			<Fragment>
+				{ baseCopy }
+				<br />
```
Since this is part of a paragraph tag, a simple `<br />` is the most suitable solution here I guess.

tyxla commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -483,6 +546,7 @@ export class PlansFeaturesMain extends Component {
 				<QueryPlans />
 				<QuerySites siteId={ siteId } />
 				<QuerySitePlans siteId={ siteId } />
+				<QuerySitePurchases siteId={ siteId } />
```
Right now purchases will only be used for Jetpack sites. Should we render this query component only in the instance where `this. isJetpackBackupAvailable()` is `true` or something similar?

Will be redundant anyway if we decide to move `getProductsSelectorSubHeader` to a component as I suggested above.

tyxla commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -445,7 +508,7 @@ export class PlansFeaturesMain extends Component {
 				<FormattedHeader
 					headerText={ translate( 'Solutions' ) }
-					subHeaderText={ translate( 'Just need backups? Learn about add-on solutions.' ) }
+					subHeaderText={ this.getProductsSelectorSubHeader() }
```
How do you feel about moving `getProductsSelectorSubHeader` to its own component? It would really help in keeping the already huge `PlansFeaturesMain` clean from any unnecessary logic.

tyxla commented:
Also cc @ChaosExAnima who is working on the Jetpack side of this - this looks like a good first PR on the Jetpack side too. Maybe we could keep the iterations on Calypso and Jetpack parallel? That would make it easier for others to jump into the effort if they like.

delawski commented:
@tyxla Thank you for your review and suggestion. It turned out that we already had a `header.jsx` file that was introduced earlier but at some point stopped being used. I've updated this sub-component and moved all the products selector header logic there.

Please have another look into the code.

a8ci18n commented on client/my-sites/plans-features-main/header.jsx:
```diff
@@ -1,24 +1,122 @@
+
+	getSubHeader() {
+		const { siteSlug, translate } = this.props;
+		const baseCopy = translate( "Just looking for a backups? We've got you covered." );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Just looking for backups? We’ve got you covered.' )` *ES Score: 6*

tyxla commented on client/my-sites/plans-features-main/header.jsx:
```diff
@@ -1,24 +1,122 @@
+			<Fragment>
+				{ baseCopy }
+				<br />
+				<ExternalLinkWithTracking
```
Seems like this link doesn't look great in `/jetpack/connect/store`:

![](https://cldup.com/RDw39Y00R5.png)

I guess the link should be changed to another color? cc @eeeeevon13 and @crunnells on the color - should we use the same as the text and underline it?

delawski replied:
Thanks for checking the connect flow - I totally forgot about it! 😅

I've used the same link style as there is for the link in the bottom of that page:

<img width="409" alt="Screenshot 2019-12-10 at 17 19 45" src="https://user-images.githubusercontent.com/478735/70547646-897ac400-1b71-11ea-9ee1-52942370cfb6.png">

delawski replied:
So it looks like this now:

<img width="569" alt="Screenshot 2019-12-10 at 17 24 02" src="https://user-images.githubusercontent.com/478735/70547850-e5dde380-1b71-11ea-9bf0-b215f27c113b.png">

@crunnells Does it look okay? Or do you want me to change the link appearance in Jetpack Connect flow?

tyxla commented on client/my-sites/plans-features-main/header.jsx:
```diff
@@ -1,24 +1,122 @@
+					icon
+				>
+					{ translate( 'Which backup option is best for me?' ) }
+				</ExternalLinkWithTracking>
```
Something to note: I think we should display this link only if the user doesn't have a product purchased yet. That's what we used to do when this link was in the previous location.

delawski replied:
I've changed the behaviour so that if a plan has a backup feature it will still get the link here.

delawski commented on client/my-sites/plans-features-main/header.jsx:
```diff
@@ -1,24 +1,122 @@
+
+		return (
+			<Fragment>
+				<QuerySitePlans siteId={ selectedSiteId } />
```
Sounds good! I've tested without it and it worked well. That's why I thought we don't need the `QueryPlans` here. I've added it, though, in the latest commit.

a8ci18n commented on client/my-sites/plans-features-main/products-header.jsx:
```diff
@@ -0,0 +1,105 @@
+
+	getSubHeader() {
+		const { siteSlug, translate } = this.props;
+		const baseCopy = translate( "Just looking for a backups? We've got you covered." );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Just looking for backups? We’ve got you covered.' )` *ES Score: 6*

tyxla replied:
Good suggestion, but FWIW we're consciously doing this change.

tyxla commented on client/my-sites/plans-features-main/products-header.jsx:
```diff
@@ -0,0 +1,105 @@
+	siteSlug: PropTypes.string,
+};
+
+export default connect( ( state, { siteId } ) => {
```
Seems like in this component we never use the `siteId` prop. Should we completely remove it and rely on the `selectedSiteId` one instead?

delawski replied:
I thought about it but in the end I took [the same approach as in the `ProductsSelector`](https://github.com/Automattic/wp-calypso/blob/7b1074f1119fdff6ce182df8f3349c2d1da49f27/client/blocks/product-selector/index.jsx#L583-L584).

I guess it is a bit more flexible, though we don't really use it at this point.

tyxla replied:
There's the difference though. For `ProductsSelector` we use that ID (even if it's only for devdocs). And for this component we don't. So I'd suggest going YAGNI and cleaning it up.

delawski commented on client/my-sites/plans-features-main/products-header.jsx:
```diff
@@ -0,0 +1,105 @@
+
+		return (
+			<Fragment>
+				{ baseCopy }
```
The whole `subheader` is wrapped with a paragraph tag. In my previous comment (that disappeared after moving the code to `products-header.jsx`) I noted that since it is inside a paragraph tag already, it makes the most sense to just use `<br />`. It's a semantic and simple solution in my opinion.

delawski commented:
@tyxla Thank you for your review. I've addressed the remaining feedback items in [the latest commit](https://github.com/Automattic/wp-calypso/pull/38311/commits/97dd6cfe781aeab6d193de22ff12ab5a1f6973b1). Please have another look.

tyxla commented:
> @tyxla Thank you for your review. I've addressed the remaining feedback items in the latest commit. Please have another look.

LGTM @delawski! Just have [one suggestion](https://github.com/Automattic/wp-calypso/pull/38311#discussion_r356595549), which is by no means blocking.

Thanks!

delawski commented:
@tyxla I've removed the `siteId` prop as you suggested. There are also 2 more changes made which I'd like you to have a look:
- I've added a loading state when the link it not displayed. I noticed that for users who have Jetpack Backup product the link was initially shown and then abruptly disappeared. It's now fixed.
- I've removed the two unused queries: `QueryPlans` and `QuerySitePlans`. We're no longer using them since we're just checking if a user has Jetpack Backup product (using `QuerySitePurchases`) and disregard plan entirely.
